101-rds-deployment-full-ha/azuredeploy.json

```diff
@@ -0,0 +1,1180 @@
+                            }
+                        ],
+                        "pubIntlbPool": [
+                            {
+                                "id": "[concat(resourceId('Microsoft.Network/loadBalancers',parameters('intLbName')),'/backendAddressPools/',parameters('intLbBackEndPool'))]"
+                            },
+                            {
+                                "id": "[concat(resourceId('Microsoft.Network/loadBalancers',parameters('pubLbName')),'/backendAddressPools/',parameters('pubLbBackEndPool'))]"
+                            }                            
+                        ],
+                        "intlbRef": "[resourceId(resourceGroup().name,'Microsoft.Network/loadBalancers',parameters('intLbName'))]"
+                    },
+                    "resources": [
+                        {
+                            "type": "Microsoft.Network/publicIPAddresses",
+                            "apiVersion": "2019-12-01",
+                            "location": "[parameters('location')]",
+                            "sku": {
+                                "name": "Standard"
+                            },
+                            "properties": {
+                                "publicIPAllocationMethod": "Static",
+                                "dnsSettings": {
+                                    "domainNameLabel": "[concat(parameters('vmName'),'pip',copyIndex(1))]"
+                                }
+                            },
+                            "name": "[concat(parameters('vmName'),copyIndex(1),'pip')]",
+                            "copy": {
+                                "name": "[concat(parameters('vmName'),'pipcopy')]",
+                                "count": "[parameters('loopCount')]"
+                            }
+                        },
+                        {
+                            "type": "Microsoft.Network/networkInterfaces",
+                            "apiVersion": "2019-12-01",
+                            "location": "[parameters('location')]",
+                            "dependsOn": [
+                                "[concat(parameters('vmName'),copyIndex(1),'pip')]"
+                            ],
+                            "properties": {
+                                "ipConfigurations": [
+                                    {
+                                        "name": "ipconfig",
+                                        "properties": {
+                                            "subnet": {
+                                                "id": "[parameters('subNetRef')]"
+                                            },
+                                            "privateIPAllocationMethod": "[if(contains(parameters('vmName'),'dc'),if(equals(copyIndex(1),1),'static','dynamic'),'dynamic')]",

```

Comment:
need a better/more deterministic condition here... the vmNames (as written) can be changes so they may not be any vm with DC in the name or all vmNames may have dc in the name..

Reply:
still needs to be addressed